src/aatoolbox/datasources/glofas/forecast.py

```diff
@@ -0,0 +1,266 @@
+"""Download and process GloFAS forecast and reforecast river discharge data."""
+import logging
+from datetime import datetime
+from pathlib import Path
+from typing import List, Union
+
+import xarray as xr
+from dateutil import rrule
+
+from aatoolbox.config.countryconfig import CountryConfig
+from aatoolbox.datasources.glofas import glofas
+from aatoolbox.utils.check_file_existence import check_file_existence
+from aatoolbox.utils.geoboundingbox import GeoBoundingBox
+
+logger = logging.getLogger(__name__)
+
+
+class _GlofasForecastBase(glofas.Glofas):
+    """Base class for all GloFAS forecast data downloading and processing."""
+
+    def __init__(
+        self,
+        country_config: CountryConfig,
+        geo_bounding_box: GeoBoundingBox,
+        leadtime_max: int,
```

Comment:
Nitpicking, but maybe have order of parameters the same as the base class. I think just `leadtime_max` is out of place.

Reply:
Don't worry I love fixing this kind of thing! :D